File: 506-relative-ranks/506-relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        ranks = []

        for i, val in enumerate(score):
            heapq.heappush(ranks, (-val, i))
        print(ranks)
        res = [0] * len(score)
        r = 1
        rank = ["Gold Medal", "Silver Medal", "Bronze Medal"]
        while len(ranks) != 0:
            _, i = heapq.heappop(ranks)
            if r <= 3:
                res[i] = rank[r - 1]
            else:
                res[i] = str(r)
            r += 1
        return res

File: 506-relative-ranks/506-relative-ranks.cpp
class Solution
{
public:
  vector<string> findRelativeRanks(vector<int> &score)
  {
    priority_queue<pair<int, int>> q;
    for (int i = 0; i < score.size(); i++)
    {
      q.push({score[i], i});
    }
    int i = 0;

    vector<string> res(size(q));

    while (!q.empty())
    {
      res[q.top().second] = i == 0 ? "Gold Medal" : i == 1 ? "Silver Medal"
                                                : i == 2   ? "Bronze Medal"
                                                           : to_string(i + 1);
      i++;
      q.pop();
    }
    return res;
  }
};